src/utils/hdnode.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @packageDocumentation
+ * @module Utils-HDNode
+ */
+
+import { Buffer } from 'buffer/'
+import hdnode from 'hdkey'
+
+/**
+ * BIP32 hierarchical deterministic keys.
+ */
+
+export default class HDNode {
+  private hdkey: any
+  publicKey: Buffer
+  privateKey: Buffer
+  chainCode: Buffer
+  privateExtendedKey: string
+  publicExtendedKey: string
+
+  /**
+  * Derives the HDNode at path from the current HDNode.
+  * @param path 
+  */
+  derive(path: string): HDNode {
+    const hdKey = this.hdkey.derive(path)
+    let hdNode: HDNode
+    if (hdKey.privateExtendedKey != null) {
+      hdNode = new HDNode(hdKey.privateExtendedKey)
+    } else {
+      hdNode = new HDNode(hdKey.publicExtendedKey)
+    }
+    return hdNode
+  }
+
+  /**
+  * Signs the buffer hash with the private key using secp256k1 and returns the signature as a buffer.
+  * @param hash 
```

Comment:
Should `@return` be added to this comment and above function `derive`'s comment, or do you think it's unnecessary? I could go either way.

Reply:
Agreed. Resolved.